pkg/sentry/platform/ring0/pagetables/pagetables.go

```diff
@@ -90,6 +110,15 @@ func (*mapVisitor) requiresSplit() bool { return true }
 //
 //go:nosplit
 func (p *PageTables) Map(addr usermem.Addr, length uintptr, opts MapOpts, physical uintptr) bool {
+	if p.upperHalfPageTables != nil {
+		// ignore change to the read-only upperhalf
+		if uintptr(addr) >= halfSize {
+			return false
+		}
+		if uintptr(addr)+length > halfSize {
```

Comment:
We should add an overflow check to the top of this function, to ensure that this is correct.